src/util/StringUtils.h

```diff
@@ -207,6 +207,27 @@ inline size_t findLiteralEnd(const std::string_view input,
   return endPos;
 }
 
+// Implementation based on https://stackoverflow.com/a/25374036
+inline bool constantTimeEqualsImpl(std::basic_string_view<volatile char> str1,
+                            std::basic_string_view<volatile char> str2) {
+  if (str1.length() != str2.length()) {
+    return false;
+  }
+  volatile char c = 0;
+  for (size_t i = 0; i < str1.length(); ++i) {
+    c = c | (str1[i] ^ str2[i]);
```

Comment:
Interesting: Compound assignment for volatile variables is deprecated in C++20 and un-deprecated in C++23.
See https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1152r0.html#compound and https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2327r1.pdf for more information